pkg/kubelet/volumemanager/reconciler/reconciler_test.go

```diff
@@ -1834,7 +1835,6 @@ func Test_Run_Positive_VolumeMountControllerAttachEnabledRace(t *testing.T) {
 	fakePlugin.UnmountDeviceHook = func(mountPath string) error {
 		// Act:
 		// 3. While a volume is being unmounted, add it back to the desired state of world
-		t.Logf("UnmountDevice called")
 		generatedVolumeName, err = dsw.AddPodToVolume(
```

Comment:
I prefer to have the callback use a local variable rather instead of changing the outer function.

Reply:
Okay, I will change it to a local variable later.

Reply:
Done. No data race. The error and cause are the same as before.

```
# ./stress -p 300 ./reconciler.test -test.run "Test_Run_Positive_VolumeMountControllerAttachEnabledRace" 
35m10s: 280289 runs so far, 16 failures (0.01%)
# grep -r "Cannot get volumes from disk" /tmp/go-stress-20210716T123104* |wc -l
16
# grep -r "DATA" /tmp/go-stress-20210716T123104* |wc -l
0
```

Reply:
go vet doesn't allow me to do this, `assignment copies lock value to tCopy: testing.T contains testing.common contains sync.RWMutex`
I will change the log output in the goroutine to klog later.

Reply:
Sorry can you clarify, are we able to have this inner function use a local variable for the `generatedVolumeName`?

Reply:
I believe this is another solution to this problem. When the `generatedVolumeName ` is not written in the hook but a local variable is used, it is impossible to cause data race for the `generatedVolumeName`.

If you prefer your solution, I can modify it. :-)